src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke.util;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+
+/**
+ * Represents a dealer to process a full command.
+ */
+public class Parser {
+    private TaskList taskList;
+    private Storage storage;
+    private Ui ui;
+
+    /**
+     * Represents a new parser for Duke.
+     * @param taskList
+     * @param ui
+     * @param storage
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses an input and process it,
+     * example of the input command should be like: deadline return book /by 2/12/2019 1800 .
+     *
+     * @param input a full command
+     */
+    public void parse(String input) {
```

Comment:
Perhaps extract out the methods for each if else block to make the current method shorter and more readable?